### PR TITLE
Fix unnecessary tokeninventory fetch

### PR DESCRIPTION
**Pull request by tiagoalvesdulce**
Closes https://github.com/decred/politeiagui/issues/1906

### Solution description

~~Really simple one-liner that fixes the tokenInventory selector but took me some time to come up with it :)~~ Turn out the solution was a bit more complex and I also improved the way we handle request actions. I'm now only requesting the tokenInventory when it's needed.

### CODE REVIEW

**Comment by thi4go** [comment]
Hey mate, so I was wondering something. Since we finished with this redux refactor (https://github.com/decred/politeiagui/issues/1490), shouldn't the selector be getting the token inventory response from the normalized models reducers? Maybe `models/proposals`? I think that by adding the response payload on `receive`, which will add data to the `api` state branch, we would be back to square one and hurt our single source of truth principle, that after the refactor, is supposed to be on `models` branch.

**Comment by tiagoalvesdulce** [comment]
Not sure if I get what you're saying but we want to check the raw tokenInventory from the API with the `allByStatus` (normalized state). The token inventory payload shouldn't be used anywhere else.

File: src/selectors/models/proposals.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ export const allByStatus = createDeepEqualSelector(
 );
 export const tokenInventory = createDeepEqualSelector(
   get(["api"]),
```
Comment (thi4go) [inline src/selectors/models/proposals.js:17]: I think that tokenInventory should fetch the data from `models/proposals`, and no the `api` state branch. Is there any specific reason it is fetching this from the `api`?

Comment (tiagoalvesdulce) [reply src/selectors/models/proposals.js:17]: Yes, there is. I want to check if both are in sync, if they aren't, I want to do a new fetch.

**Pull request merged**